src/enrichment_tables/pgtable.rs

```diff
@@ -0,0 +1,440 @@
+    // #[snafu(display("host missing"))]
+    // HostMissing,
+    // #[snafu(display("multiple hosts not supported: {:?}", hosts))]
+    // MultipleHostsNotSupported { hosts: Vec<Host> },
+}
+
+/// Configuration of TLS when connecting to PostgreSQL.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+struct PgtableTlsConfig {
+    /// Absolute path to an additional CA certificate file.
+    ///
+    /// The certificate must be in the DER or PEM (X.509) format.
+    #[configurable(metadata(docs::examples = "certs/ca.pem"))]
+    ca_file: PathBuf,
+}
+
+/// Postgres-specific settings
+#[configurable_component(enrichment_table("pgtable"))]
+#[derive(Clone, Debug, Default)]
+pub struct PgtableConfig {
+    #[configurable(derived)]
+    /// The postgres connection string
+    endpoint: String,
+    /// Table to query
+    table: String,
+    /// Columns to project from table
+    columns: Vec<String>,
+    /// TLS client certificate
+    #[configurable(derived)]
+    tls: Option<PgtableTlsConfig>,
+}
+impl_generate_config_from_default!(PgtableConfig);
+
+fn surround_with_quotes(name: String) -> String {
+  return ["\"", &name, "\""].concat();
+}
+
+fn row_to_vec_value(row: Row) -> Vec<Value> {
+    let row_len = row.len();
+    let mut dst: Vec<Value> = Vec::new();
+    for i in 0..row_len {
+      let rint : Result<i64, PgError> = row.try_get(i);
+      match rint {
+        Err(_) => {
+          let rstr : Result<String, PgError> = row.try_get(i);
+          match rstr {
+            Ok(s) => {
+              dst.push(Value::Bytes(Bytes::from(s)));
+            }
+            Err(_) => {
+              dst.push(Value::Null);
+            }
+          }
+        }
+        Ok(i) => {
+          dst.push(Value::Integer(i));
+        }
+      }
+    };
+    return dst;
+}
+
+impl EnrichmentTableConfig for PgtableConfig {
+    async fn build(
+        &self,
+        _globals: &crate::config::GlobalOptions,
+    ) -> crate::Result<Box<dyn Table + Send + Sync>> {
+        let config: Config = self.endpoint.parse()?;
+        let client = match &self.tls {
+            Some(tls_config) => {
+                let mut builder = SslConnector::builder(SslMethod::tls_client())?;
+                builder.set_ca_file(tls_config.ca_file.clone())?;
+                let connector = MakeTlsConnector::new(builder.build());
+
+                let (client, connection) = config.connect(connector).await?;
+                tokio::spawn(connection);
+                client
+            }
+            None => {
+                let (client, connection) = config.connect(NoTls).await?;
+                tokio::spawn(connection);
+                client
+            }
+        };
+        let columns_with_commas : Vec<String> = intersperse(self.columns.clone().into_iter().map(surround_with_quotes), ",".to_string()).collect();
+        let select_query : String = ["SELECT ", &columns_with_commas.concat(), " FROM ", &self.table].concat();
```

Comment:
We need to make sure than an enrichment table config along the lines of:

```yaml
table: table; DROP TABLE table
```

results in an error.

Reply:
I'm not sure what the best way to sanitize these is. It's not possible to use prepared statements for this. I could just restrict the character set to something like alphanumeric and underscore. Is there a better way to do this with `tokio_postgres`?

Reply:
I am in the fortunate position to have not had to do much SQL for the past few years. If I recall I think it is possible to quote the table name with an arbitrary number of quotes.

So if table name was `table; DROP TABLE table` we would do: 

```sql
select * from 'table; DROP TABLE table'
```

And if table name was `table'; DROP TABLE table`, I think it is possible to use two quotes in the query. So

```sql
select * from ''table'; DROP TABLE table''
```

If table name was `table''; DROP TABLE table`:

```sql
select * from '''table''; DROP TABLE table'''
```

And so on...

Reply:
For example, the databend sink, does not block this kind of behaviour https://github.com/vectordotdev/vector/blob/695bb476d6116ba65d57a77aa0ac655af76f063a/src/sinks/databend/service.rs#L133

If this is specified in the config... I think it should be user's task to ensure that there is no injection